pkg/provider/cloud/gke/provider.go

```diff
@@ -479,3 +481,31 @@ func createClient(ctx context.Context, serviceAccount string, scope string) (*ht
 
 	return client, projectID, nil
 }
+
+func DecodeError(err error) error {
+	if err == nil {
+		return nil
+	}
```

Comment:
This nil check can be removed, I think. errors.As returns false if the error is nil: https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/errors/wrap.go;l=91